### PR TITLE
Comment improvements on SSLConfigProvider reference.conf

### PR DESCRIPTION
**Pull request by ignasi35**

### CODE REVIEW

**Review by raboof — APPROVED** [review]
Good info!

File: akka-remote/src/main/resources/reference.conf [hunk 1]
```diff
@@ -659,13 +659,15 @@ akka {
...
         # TLS 1.2 is available since JDK7, and default since JDK8:
         # https://blogs.oracle.com/java-platform-group/entry/java_8_will_use_tls
+        # TLS 1.3 is available since JDK 11:
+        # https://www.oracle.com/technetwork/java/javase/11-relnote-issues-5012449.html
         protocol = "TLSv1.2"
 
         # Example: ["TLS_RSA_WITH_AES_128_CBC_SHA", "TLS_RSA_WITH_AES_256_CBC_SHA"]
-        # You need to install the JCE Unlimited Strength Jurisdiction Policy
-        # Files to use AES 256.
+        # If you use a JDK 7 prior to 7u171 or a JDK 8 prior to 8u161 you need to install
```
Comment (johanandren) [inline akka-remote/src/main/resources/reference.conf:667]: We don't support pre JDK 8 (because Scala) so that part is probably not worth mentioning?

**Review by johanandren — APPROVED** [review]

File: akka-remote/src/main/resources/reference.conf (hunk 1 above)
Comment (ignasi35) [reply akka-remote/src/main/resources/reference.conf:667]: I wasn't entirely sure so err'd to the safe side. Thanks for alerting!

File: akka-remote/src/main/resources/reference.conf [hunk 2]
```diff
@@ -659,13 +659,15 @@ akka {
         # Protocol to use for SSL encryption, choose from:
         # TLS 1.2 is available since JDK7, and default since JDK8:
         # https://blogs.oracle.com/java-platform-group/entry/java_8_will_use_tls
+        # TLS 1.3 is available since JDK 11:
+        # https://www.oracle.com/technetwork/java/javase/11-relnote-issues-5012449.html
```
Comment (patriknw) [inline akka-remote/src/main/resources/reference.conf:663]: do we dare to say that we support that without any testing?

Comment (ignasi35) [reply akka-remote/src/main/resources/reference.conf:663]: Defintely not (especially considering https://github.com/akka/akka/issues/28092#issuecomment-547743545).

Comment (ignasi35) [reply akka-remote/src/main/resources/reference.conf]: (slightly off-topic) TIL, ["TLS 1.3 is not directly compatible with previous versions"](https://www.oracle.com/technetwork/java/javase/11-relnote-issues-5012449.html) so if/when we want to support it we may need to find a way to support rolling upgrades.

**Review by johanandren — APPROVED** [review]

**Pull request merged**